docs/features/software-catalog/external-integrations.md

```diff
@@ -171,3 +171,98 @@ export default async function createPlugin(
 
 Start up the backend - it should now start reading from the previously
 registered location and you'll see your entities start to appear in Backstage.
+
+## Caching processing results
+
+The catalog periodically refresh entities in the catalog and by doing so it call
+out to external systems to fetch changes. This can be taxing for upstream
+services and large deployments may get rate limited if too many requests are
+sent. Luckily many external systems provide ETag support to check for changes
+which usually don't count towards the quota and saves resources both internally
+and externally.
+
+The catalog has built in support for caching when refreshing external locations
+in GitHub. This example aims to demonstrate the same behavior for `system-x`
+that we implemented earlier.
+
+```ts
+import { UrlReader } from '@backstage/backend-common';
+import { Entity, LocationSpec } from '@backstage/catalog-model';
+import {
+  results,
+  CatalogProcessor,
+  CatalogProcessorEmit,
+  CatalogProcessorCache,
+} from '@backstage/plugin-catalog-backend';
+import { CatalogProcessorParser } from '.';
+
+// It's recommended to always bump the CACHE_KEY version if you make
+// changes to the processor implementation or CacheItem.
+const CACHE_KEY = 'v1';
+
+// CacheItem is our cache containing ETag used in the upstream request
+// as well as the processing result used when the Etag matches.
+// Bump the CACHE_KEY version if you make any changes to this type.
+type CacheItem = {
+  etag?: string;
```

Comment:
Think it makes sense to have this be required and check for the existence of an etag in the reponse

Reply:
Sure thing, fixed!